src/hooks/use-scroll-percentage/index.ts

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from 'react'
+import useWindowSize from 'hooks/use-window-size'
+import {
+	getPercentageScrolled,
+	isBrowser,
+	round,
+	useDocumentScrollHeight,
+} from './helpers'
+
+const DECIMAL_PRECISION = 2
+const CUSTOM_INITIAL_EVENT_NAME = 'use-scroll-percentage_initial-update'
+
+/**
+ * Track the percentage of scrolled distance on the page.
+ *
+ * Note: the return value will be `undefined` before the first calculation
+ * of scroll percentage.
+ *
+ * Note: this hook accepts an option `mutationTargetSelector` string.
+ * If this option is used, and yields a valid element, we attach a mutation
+ * observer to that element and monitor the element and its subtree for
+ * changes. This ensures better accuracy of the `documentScrollHeight` we
+ * use to calculate scroll percentage, as in many views, document height
+ * may change significantly through user interactions that update the DOM.
+ *
+ * For example, on the Tutorial view, the contents in the `main` element
+ * frequently change height from interactions such as changing tabbed content.
+ * For that view, we pass `main` as the `mutationTargetSelector`.
+ */
+export default function useScrollPercentage({
```

Comment:
This hook is used by the new `ScrollProgress` component in `SidebarSidecarLayout`. One interesting facet is accepting a `mutationTargetSelector` here, there's more detail on why this feels necessary in comments, and in `use-document-scroll-height`.

Have tried to build this in a way that might be re-usable, for example within `use-scroll-progress-analytics`. But leaving that abstraction & re-use step for a separate PR.